### PR TITLE
Add parser/detector for CodeQL SARIF files

### PR DESCRIPTION
**Pull request by drdavella**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -88,6 +88,7 @@ defectdojo = "core_codemods:defectdojo_registry"
 
 [project.entry-points.sarif_detectors]
 "semgrep" = "codemodder.semgrep:SemgrepSarifToolDetector"
+"codeql" = "codemodder.codeql:CodeqlSarifToolDetector"
```
Comment (bdoyal) [inline pyproject.toml]: `CodeQL` or `Codeql`?

File: src/codemodder/codemods/codeql.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class CodeQLSarifFileDetector(BaseDetector):
+    def apply(
+        self,
+        codemod_id: str,
+        context: CodemodExecutionContext,
+        files_to_analyze: list[Path],
+    ) -> ResultSet:
+        del codemod_id
+        del files_to_analyze
```
Comment (bdoyal) [inline src/codemodder/codemods/codeql.py:18]: What is the significance of del here?

**Review by bdoyal — APPROVED** [review]

File: src/codemodder/codemods/codeql.py (hunk 2 above)
Comment (drdavella) [reply src/codemodder/codemods/codeql.py:18]: It makes the type checker happy by indicating that these variables are unused.

File: pyproject.toml (hunk 1 above)
Comment (drdavella) [reply pyproject.toml]: -__-

**Review by clavedeluna — APPROVED** [review]

**Pull request merged**